cleanlab/data_valuation.py

```diff
@@ -113,4 +136,8 @@ def data_shapley_knn(
     # Use provided knn_graph or compute it from features
     if knn_graph is None:
         knn_graph, _ = create_knn_graph_and_index(features, n_neighbors=k, metric=metric)
-    return _knn_shapley_score(knn_graph, labels, k)
+
+    num_examples = labels.shape[0]
+    distances = knn_graph.indices.reshape(num_examples, -1)
+    scores = _knn_shapley_score(neighbor_indices=distances, y=labels, k=k)
+    return 0.5 * (scores + 1)
```

Comment:
youre confident this is better than doing:
`max(scores, 0)` ?

Reply:
what's the smallest raw (pre-transformation) score you've encountered?

Reply:
or what about: 

```
(x+1) * 0.001 if x <= 0
0.001 + x*(1 - 0.001) if x > 0
```

Reply:
For smaller random data points, I've observed values closer to the extreme ranges of [-1, 1].

I also believe that for larger datasets, the raw scores tend to shrink around zero, essentially scaling themselves down. However, I would discourage attempting to rescale these values, as we have limited control over them.

The `0.5(x +1)` transformation is good enough to stick to our canonical [0, 1] range.

Reply:
The initial suggestion to use `max(score, 0)` was considered well before the latest revision of the raw score implementation.

Using `0.5 * (scores + 1)` is more effective than max(scores, 0) because it accounts for both positive and negative scores, transforming them consistently into the [0, 1] range.

`0.5 * (scores +1)` is better than `max(scores, 0)` as the positive scores may also be exceedingly small.
Then we risk againt that someone would interpret it as all points have small scores which is an issue.

The `max(score, 0)` approach might lead to misinterpretations, especially when positive scores are very small, potentially conveying that all points have low scores, which is not the case.

We'll have a more accurate and balanced representation of the scores within our desired range if we stick to `0.5 * (scores + 1)`.

Reply:
sg thanks for the info!